website/content/docs/v0.14/Learn More/developing-talos.md

```diff
@@ -0,0 +1,225 @@
+---
+title: "Developing Talos"
+weight: 13
+---
+
+This guide outlines steps and tricks to develop Talos operating systems and related components.
+The guide assumes Linux operating system on the development host.
+Some steps might work under Mac OS X, but using Linux is highly advised.
+
+## Prepare
+
+Check out the [Talos repository](https://github.com/talos-systems/talos).
+
+Try running `make help` to see available `make` commands.
+You need Docker and `buildx` installed on the host.
+
+> Note: On Ubuntu 21.10 it's easier to use Ubuntu's `docker`, for earlier versions of Ubuntu use [Docker's apt repositories](https://docs.docker.com/engine/install/ubuntu/).
+>
+> `buildx` is not available with Ubuntu packages, it should be installed [as a plugin from GitHub releases](https://docs.docker.com/buildx/working-with-buildx/#install).
+
+Set up a builder with access to host network:
+
+```bash
+ docker buildx create --driver docker-container  --driver-opt network=host --name local1 --buildkitd-flags '--allow-insecure-entitlement security.insecure' --use
+```
+
+> Note: `network=host` allows buildx builder to access host network, so that it can push to a local container registry (see below).
+
+Make sure the following steps work:
+
+- `make talosctl`
+- `make initramfs kernel`
+
+Set up a local docker registry:
+
+```bash
+docker run -d -p 5005:5005 \
+    --restart always \
+    --name local registry:2
+```
+
+Try to build and push to local registry an installer image:
+
+```bash
+make installer IMAGE_REGISTRY=127.0.0.1:5005 PUSH=true
```

Comment:
I think it also worths mentioning that you can pin the tag.
Not a big deal, but it's easier sometimes if you don't want to copy/paste the full image path everytime.